// Check that predefined exceptions are properly thrown.
// Some were regressions.

// Bug report from Alexandre Gabus.
var test;
try
{
  test += 3;
}
catch
{
  test = 1;
  test += 3;
};
[00000001] 4

echo(test);
[00000002] *** 4


// while in asTopLevelPrintable.
class X { function asString () { 0 / 0 }};
[00000003:error] !!! input.u:@.34-38: /: division by 0


1:1;
[00027095:error] !!! input.u:@.1-3: unexpected 1, expected a Tag


// Issue #4684.  The kernel remains stuck because the scheduler does
// not wake up the job in charge.
try
{
  { sleep(1) | echo("THROW") | throw 1 },
  t: { t.freeze }
}
catch (var e)
{
  1;
};
[00000005] *** THROW
[00000006] 1

2;
[00000007] 2
